packages/adminrouter/extra/src/test-harness/modules/mocker/endpoints/recording.py

```diff
@@ -77,6 +77,12 @@ def _process_commands(self, blob):
             self._record_request()
             # Recording does not end the request processing, so we do not
             # return anything here
+        if ctx.data["encoded_response"]:
+            msg_fmt = "Endpoint `%s` sending encoded response `%s` as requested"
+            log.debug(
+                msg_fmt, ctx.data['endpoint_id'], ctx.data["encoded_response"])
+            self._finalize_request(
+                200, 'text/plain; charset=utf-8', ctx.data["encoded_response"])
```

Comment:
We should be returning True/False as per https://github.com/dcos/dcos/blob/master/packages/adminrouter/extra/src/test-harness/modules/mocker/endpoints/basehandler.py#L81-L84 Otherwise request processing will continue while the response has already been sent.